partiql-lang/src/test/kotlin/org/partiql/lang/eval/visitors/inferencer/InferencerSimpleCaseWhenTests.kt

```diff
@@ -0,0 +1,511 @@
+                        ),
+                        createDataTypeMismatchError(
+                            SourceLocationMeta(
+                                4L,
+                                30L,
+                                17L
+                            ),
+                            argTypes = listOf(StaticType.INT.asNullable(), StaticType.SYMBOL.asOptional()),
+                            nAryOp = "CASE"
+                        )
+                    )
+                )
+            ),
+            // Todo: Inferencer keeps all the `THEN` expr types even if we know the comparison will not succeed, why?
+            TestCase(
+                name = "null or missing warning (caseValue = null): CASE <null> WHEN <string> THEN <string> WHEN <symbol> THEN <symbol> END",
+                originalSql = """
+                    CASE t_null
+                        WHEN t_string THEN t_string
+                        WHEN t_symbol THEN t_symbol
+                    END
+                    """,
+                globals = mapOf(
+                    "t_null" to StaticType.NULL,
+                    "t_string" to StaticType.STRING,
+                    "t_symbol" to StaticType.SYMBOL
+                ),
+                handler = expectQueryOutputType(
+                    expectedType = StaticType.unionOf(StaticType.STRING, StaticType.SYMBOL, StaticType.NULL),
+                    expectedWarnings = listOf(
+                        createReturnsNullOrMissingWarning(
+                            SourceLocationMeta(
+                                2L,
+                                26L,
+                                6L
+                            )
+                        )
+                    )
+                )
+            ),
+            TestCase(
+                name = "always missing error (caseValue = missing): CASE <missing> WHEN <string> THEN <string> WHEN <symbol> THEN <symbol> END",
+                originalSql = """
+                    CASE t_missing
+                        WHEN t_string THEN t_string
+                        WHEN t_symbol THEN t_symbol
+                    END
+                    """,
+                globals = mapOf(
+                    "t_missing" to StaticType.MISSING,
+                    "t_string" to StaticType.STRING,
+                    "t_symbol" to StaticType.SYMBOL
+                ),
+                handler = expectSemanticProblems(
+                    expectedErrors = listOf(
+                        createReturnsMissingError(
+                            SourceLocationMeta(
+                                2L,
+                                26L,
+                                9L
+                            )
+                        )
+                    )
+                )
+            ),
+            // Todo: Inferencer keeps all the `THEN` expr types even if we know the comparison will not succeed, why?
+            //  Missing Type is lost here, why?
```

Comment:
In my mental model, in case of that the value expression is missing, I assumed the expression should return `missing`. 

It is almost like
```
when(v) { // if missing, we failed at this stage without even going to the branch. 
   'case1' -> {...}
   'case2' -> {...}
   else -> {...}
}
```

Reply:
From other comment, https://github.com/partiql/partiql-lang-kotlin/pull/1248/files#r1364542871, think we can just opt to leave it as it is for now and keep the TODO comment. We can always adjust to make the output type "smarter" if a use case arises.